src/main/java/io/opensaber/views/Transformer.java

```diff
@@ -32,21 +32,22 @@ public JsonNode transform(ViewTemplate viewTemplate, ObjectNode node) {
             if (functionStr != null) {
 
                 String fdName = field.getFunctioName();
-                String expression = viewTemplate.getExpression(fdName);
-
-                FieldFunction function = new FieldFunction(expression);
+                FunctionDefinition funcDef = viewTemplate.getFunctionDefinition(fdName);            
+                                
                 List<Object> actualValues = new ArrayList<>();
                 for (String oneArg : field.getArgNames()) {
                     // Cut off the $
                     actualValues.add(ValueType.getValue(nodeAttrs.get(oneArg.substring(1))));
                 }
-                function.setArgValues(actualValues);
-
-                FunctionEvaluator evaluator = new FunctionEvaluator(function);
-
+                
+                IEvaluator<Object> evaluator = EvaluatorFactory.getInstance(funcDef, actualValues);
                 if (field.getDisplay()) {
-                    result.put(field.getTitle(), evaluator.evaluate().toString());
-
+                    Object evaluatedValue = evaluator.evaluate();
+                    if(evaluatedValue instanceof String){
```

Comment:
Did you test that `JsonNodeFactory.instance.pojoNode(evaluatedValue));` will not work for String? Asking if we should really have if-else here.

Reply:
Have tested this, does not work for String. Hence I preferred  a check.